urlshort/handler.go

```diff
@@ -33,5 +42,24 @@ func MapHandler(pathsToUrls map[string]string, fallback http.Handler) http.Handl
 // a mapping of paths to urls.
 func YAMLHandler(yml []byte, fallback http.Handler) (http.HandlerFunc, error) {
 	// TODO: Implement this...
-	return nil, nil
+	parsedYaml, err := parseYAML(yml)
+	if err != nil {
+		return nil, err
+	}
+	pathMap := buildMap(parsedYaml)
+	return MapHandler(pathMap, fallback), nil
+}
+
+func parseYAML(yml []byte) (out []map[string]string, err error) {
```

Comment:
I'm not sure you need this function since it would only be two lines in the **YAML**Handler function.

Reply:
The parseYAML stub was suggested in the problem. That's why I went ahead with it.